core/sql/exp/ExpLOBaccess.h

```diff
@@ -515,7 +515,7 @@ class ExLob
 
   public:
 
-    char lobDataFile_[MAX_LOB_FILE_NAME_LEN];
+    string lobDataFile_; // TODO: change to NAString when ExLobCursor is allocated off of an NAHeap
```

Comment:
Actually, it's not. See line 2338 (new file numbering):

	  //lobPtr = new (lobGlobals->getHeap())ExLob();
	  lobPtr = new ExLob();

It appears that someone experimented with putting it on the executor heap and then retreated.